### DISCUSSION TITLE
Is it possible to catch error in vue router guard when next not found route

### DISCUSSION THREAD

**Discussion by maxkopych** [vuejs/router#1885] [Help and Questions] [answered]
### Vue version

3.3.0

### Link to minimal reproduction

https://stackblitz.com/edit/vitejs-vite-6r8s84?file=src%2Frouter%2Findex.js

### Steps to reproduce

So I'm trying to catch error when `next` doesn't exist in routes
```
{
    path: "some-path",
    name: "PathName",
    component: Component,
    beforeEnter(to, from, next) {
        try {
            return next({name: "NotExistingName"})
        } catch (e) {
            return next();
        }
    }
},
```

### What is expected?

throw and error for try catch

### What is actually happening?

just warning in console

**Comment by posva** [comment, marked as answer]
You can check with `router.resolve({ name: 'route' })` to see if it exists

**Comment by maxkopych** [comment]
@posva thank you. I tried `resolve` before but I didn't put it in try catch statement and script was just dying without giving me any error.  :/ 
Code below worked for me
```
try {
  router.resolve({ name: 'NotExistingName' });
  return next({ name: 'NotExistingName' });
} catch (e) {
  return next();
}
```
